### PR TITLE
Ref/hide storages

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |Y|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Remove the getAdapters (and add the functionalities that was implemented in the core bundle) => centralize access to storage adapters

### CODE REVIEW

File: src/Storage/StorageManager.php [hunk 1]
```diff
@@ -191,4 +180,137 @@ public function initUploadFile(string $fileHash, int $fileSize, string $fileName
...
+        return $statuses;
+    }
+
+    public function getSize(string $hash): int
+    {
+        foreach ($this->adapters as $adapter) {
+            try {
+                return $adapter->getSize($hash);
+            } catch (\Throwable $e) {
+            }
```
Comment (Davidmattei) [inline src/Storage/StorageManager.php:221]: empty catch blocks look weird, maybe just continue
```suggestion
            } catch (\Throwable $e) {
                  continue;
            }
```

File: src/Storage/StorageManager.php [hunk 2]
```diff
@@ -191,4 +180,137 @@ public function initUploadFile(string $fileHash, int $fileSize, string $fileName
...
+        foreach ($this->adapters as $adapter) {
+            if ($adapter->isReadOnly()) {
+                continue;
+            }
+            try {
+                if ($adapter->remove($hash)) {
+                    ++$count;
+                }
+            } catch (\Throwable $e) {
+            }
```
Comment (Davidmattei) [inline src/Storage/StorageManager.php:312]:
```suggestion
            } catch (\Throwable $e) {
                 continue;
            }
```

**Review by Davidmattei — CHANGES_REQUESTED** [review]
New files, we can/must add declare(strict_types=1);

**Review by Davidmattei — APPROVED** [review]

File: src/Storage/StorageManager.php (hunk 1 above)
Comment (theus77) [reply src/Storage/StorageManager.php:221]: treated in the usage PR

File: src/Storage/StorageManager.php (hunk 2 above)
Comment (theus77) [reply src/Storage/StorageManager.php:312]: treated in the usage PR

**Pull request merged**